src/main/java/org/killbill/billing/plugin/api/notification/PluginTenantConfigurableConfigurationHandler.java

```diff
@@ -60,8 +60,13 @@ protected void configure(@Nullable final UUID kbTenantId) {
     public C getConfigurable(@Nullable final UUID kbTenantId) {
         // Initial configuration
         if (kbTenantId != null && !configuredTenants.contains(kbTenantId)) {
-            configure(kbTenantId);
-            configuredTenants.add(kbTenantId);
+            // Make sure to initialize the value for the tenant once
+            synchronized (configuredTenants) {
```

Comment:
I wonder how much contention this would add, since this method is used pretty often (multiple times per request, right?).
Maybe we could try the double check pattern here?
```java
    if (!configuredTenants.contains(kbTenantId)) {
        synchronized (configuredTenants) {
            if (!configuredTenants.contains(kbTenantId)) {
                ...
```
I've seen ways that claim this pattern works, but it usually for instantiating a singleton `volatile` object. No idea if it would actually work for inserting on a Set.